crates/sui-indexer-alt-jsonrpc/src/data/pg_reader.rs

```diff
@@ -129,10 +138,19 @@ impl Connection<'_> {
         Pg: QueryMetadata<Q::SqlType>,
         ST: 'static,
     {
-        debug!("{}", diesel::debug_query(&query));
+        let query_debug = diesel::debug_query(&query).to_string();
+        debug!("{}", query_debug);
+        let timer = self.metrics.db_latency.start_timer();
 
-        let _guard = self.metrics.db_latency.start_timer();
         let res = query.get_results(&mut self.conn).await;
+        let elapsed_seconds = timer.stop_and_record();
+        if elapsed_seconds > 60.0 {
```

Comment:
hardcoding itself is not ideal -- however after looking around, I felt it even worse to let all endpoints in the middle to carry extra arg / config.

Reply:
You don't need to make all the endpoints pass the config around. You can have `PgReader` accept the configuration when it is created.